### PR TITLE
Feat/#3

### PR DESCRIPTION
**Pull request by dinhdunghh333**
#3  Please comment and feedback for me, cả lò nhà em cảm ơn anh :))

### CODE REVIEW

File: src/models/types/UserAnswer.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import {Document, ExtractMongooseArray} from 'mongoose'
```
Comment (nc-minh) [inline src/models/types/UserAnswer.ts:1]: remove ExtractMongooseArray if unused

File: src/models/types/Question.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import Answer from './Answer';
+
+export default interface Question extends Document {
+    content: string;
+    image: string;
+    subject: Subject;
+    correctAnswer: Answer;
+    answers: Answer[];
+    isDeleted: boolean;
+    accuracy: Enumerator;
```
Comment (nc-minh) [inline src/models/types/Question.ts:12]: Enumerator ?

File: src/models/types/User.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import {Document} from 'mongoose';
+
+export default interface User extends Document {
+    fullname: string;
+    email: string;
+    isBlocked: boolean;
+    roles: Enumerator;
```
Comment (nc-minh) [inline src/models/types/User.ts:7]: Enumerator ?

File: src/models/types/Question.ts (hunk 2 above)
Comment (nc-minh) [reply src/models/types/Question.ts:12]: You must create a enum for accuracy like that:
enum Accuracy{
    Absolute,
    Relatively
}

File: src/models/schemas/User.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { model, Model, Schema } from 'mongoose';
+import {MODELS} from 'utils/constants/models';
+import User from 'models/types/User';
+
+const UserSchema = new Schema<User>(
+    {
+        fullname: { type: String, required: true},
+        email: {type: String, required: true, unique: true},
+        isBlocked: {type: Boolean, default: false},
+        roles: {type: Enumerator},
```
Comment (nc-minh) [inline src/models/schemas/User.ts:10]: need edit

Comment (nc-minh) [reply src/models/schemas/User.ts:10]: enum Roles{
    ADMIN,
    APPROVER,
    USER
}

**Review by nc-minh — CHANGES_REQUESTED** [review]

**Comment by dinhdunghh333** [comment]
I edited about my code'error, please check for me

**Review by nc-minh — APPROVED** [review]

**Pull request merged**